IPython/core/oinspect.py

```diff
@@ -106,21 +103,55 @@ def pylight(code):
 #****************************************************************************
 # Auxiliary functions and objects
 
-# See the messaging spec for the definition of all these fields.  This list
-# effectively defines the order of display
-info_fields = ['type_name', 'base_class', 'string_form', 'namespace',
-               'length', 'file', 'definition', 'docstring', 'source',
-               'init_definition', 'class_docstring', 'init_docstring',
-               'call_def', 'call_docstring',
-               # These won't be printed but will be used to determine how to
-               # format the object
-               'ismagic', 'isalias', 'isclass', 'found', 'name'
-               ]
+
+class InfoDict(TypedDict):
+    type_name: Optional[str]
+    base_class: Optional[str]
+    string_form: Optional[str]
+    namespace: Optional[str]
+    length: Optional[str]
+    file: Optional[str]
+    definition: Optional[str]
+    docstring: Optional[str]
+    source: Optional[str]
+    init_definition: Optional[str]
+    class_docstring: Optional[str]
+    init_docstring: Optional[str]
+    call_def: Optional[str]
+    call_docstring: Optional[str]
+    subclasses: Optional[str]
+    # These won't be printed but will be used to determine how to
+    # format the object
+    ismagic: bool
+    isalias: bool
+    isclass: bool
+    found: bool
+    name: str
+
+
+info_fields = list(InfoDict.__annotations__.keys())
+
+
+@dataclass
+class InspectorHookData:
+    """Data passed to the mime hook"""
+
+    obj: Any
+    info: Optional[OInfo]
+    info_dict: InfoDict
+    detail_level: int
+    omit_sections: list[str]
 
 
+@undoc
 def object_info(**kw):
-    """Make an object info dict with all fields present."""
-    infodict = {k:None for k in info_fields}
+    """DEPRECATED: Make an object info dict with all fields present."""
```

Comment:
Maybe it should not get deprecated because it is used on an unhappy path in IPython codebase:

https://github.com/ipython/ipython/blob/6f35f84e95c16700f9c185a962e5006a584581bb/IPython/core/interactiveshell.py#L1862-L1871

I still _think_ it is a utility which does not need to be documented/public.

Reply:
Resolved in https://github.com/ipython/ipython/pull/14342/commits/6154002c46c92b0b8750930cc4411ceaf1facc8d

Reply:
When I'm unsure about deprecating something I do a `PendingDeprecationWarning` without a date/version and point to an issue for discussing (non)-removal. Like [this]
(https://github.com/ipython/ipython/blob/b479c4a03bb4112c6aeea262a99c0bcbb18bdfa0/IPython/utils/coloransi.py#L123-L131) for example.